packages/polaris-viz/src/components/StackedAreaChart/components/StackedAreas/StackedAreas.tsx

```diff
@@ -84,8 +86,14 @@ export function StackedAreas({
   return (
     <Fragment>
       {stackedValues.map((data, index) => {
+        const dataIsValidForAnimation =
+          !previousStackedValues ||
+          data.length === previousStackedValues[index].length;
```

Comment:
This is a pattern I borrowed from `<LineSeries />` ([link to code])(https://github.com/Shopify/polaris-viz/blob/main/packages/polaris-viz-core/src/components/LineSeries/LineSeries.tsx#L111-L114). I wasn't able to figure out a way to get react spring to not animate when the data length changes but I'm open to ideas if folks know of a better way that creating a whole new component 🙏